text/0258-env-context-baggage-carriers.md

```diff
@@ -0,0 +1,359 @@
+# Environment Variables as Context and Baggage Carriers
+
+Add requirements for inter-process context and baggage propagation using
+environment variables as the carrier.
+
+This OTEP is based on discussion in [opentelemetry-specification #740](https://github.com/open-telemetry/opentelemetry-specification/issues/740#issue-665588273), builds on the [Context Propagators
+API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md)
+and follows [OTEP 0205: Context propagation requirements for messaging semantic conventions](https://github.com/open-telemetry/oteps/blob/main/text/trace/0205-messaging-semantic-conventions-context-propagation.md)
+as a guide.
+
+## Motivation
+
+There are existing requirements for context propagation over HTTP,
+messaging systems, and others, but there are no requirements for
+inter-process context propagation, although there is considerable
+ad-hoc propagation being done using environment variables today (see [Prior Art](#prior-art)).
+This OTEP is to formalize a standard based on the existing informal
+consensus.
+
+These are some examples of batch and system use cases that could
+benefit from this standardization:
+
+* CI/CD systems
+* Build and deployoment tooling
+* ETL
+* Automation tools
+* System-level tooling
+* Command-line tools, scripts, etc.
+* Tracing helper tools
+
+## Explanation
+
+To propagate context from a parent process to a child process,
+instrumentation can use environment variable(s) with upper-cased variable names
+as a carrier for the TextMapPropagator:
+
+```
+  +----------------+
+  | Parent Process |
+  +----------------+
+          |
+          | Environment variable(s) used as the carrier
+          | for context propagation, e.g.: TRACEPARENT
+          v
+  +----------------+
+  | Child Process  |
+  +----------------+
+```
+
+## Proposed addition to Process and Process Runtime Resources Semantic Conventions
```

Comment:
I am not following how propagation format is in any way related to semantic conventions and resources.

Reply:
I agree. This should be `specification` imo.